File: grammerpractice/src/main/java/javaarray/ArrayDemo.java
package javaarray;

/**
 * 数组练习
 * 1.数组必须指定长度（或者是定长）
 * 2.注意数组角标越界
 */
public class ArrayDemo {
    public static void main(String[] args) {
        //int数组
        int[] ints = new int[4];
        ints[0]=0;
        ints[1]=1;
        ints[2]=2;
        ints[3]=3;

        System.out.println(ints.length);
        for(int i=0;i<ints.length;i++){
            int value=ints[i];
            System.out.println(value);
        }

        //String 数组
        String[] strings ={"chenyanning","weiyi","shenzhen"};
        for (String str:strings
             ) {
            System.out.println(str);
        }



    }
}

File: grammerpractice/src/main/scala/scalaarray/ArrayDemo.scala
package scalaarray

object ArrayDemo {
  def main(args: Array[String]): Unit = {
    val array = new Array[String](3)
    array(0)="1";
    array(1)="2";

    val ints = new Array[Int](8)
    ints(0)=9;
    ints(1)=1;
    ints(2)=5;
    ints(3)=6;
    ints(4)=2;



    var array1=Array("fsfsa","chenyanning");
    //输出数组元素
    for (x<-array){
      println(x)
    }

    //求最大数据元素
    var tmp=0;
    for(x<- ints){
      if(tmp<x){
        tmp=x;
      }
    }
    println("tmp:"+tmp);

    var tmp1=0;
    for(i <-0 to (ints.length-1)){
      if (tmp1<ints(i)) tmp1=ints(i);
    }
    println("tmp1:"+tmp1)



  }

}
